src/ui.tsx

```diff
@@ -357,7 +387,35 @@ export class UserInterface extends Component<Record<string, never>, State> {
   };
 
   changeSlice = (e: ChangeEvent, value: number): void => {
-    this.setState({ sliceIndex: value, imageData: this.slicesData[value] });
+    this.setState(
+      {
+        sliceIndex: value,
+      },
+      () => {
+        this.mixChannels().catch((error) => {
```

Comment:
it's maybe better to catch this in mixChannels, rather then where we use it (if we do it multiple times  and don't really do anything on error)

#L323 could do:

```typescript
try {
   const displayedImage = await createImageBitmap(canvas);
   this.setState({ displayedImage });
} catch(e) {
  console.log(e);
}
```
as I don't know what happens if createImageBitmap fails here:
```typescript
this.setState({ displayedImage: await createImageBitmap(canvas) });
```
 and means only 1 console.log.
 
`this.setState({  sliceIndex: value }, this.mixChannels);` should work too and is a little shorter

Reply:
Although getting typescript error "Promise returned in function argument where a void return was expected" when removing the catch at line 395. Could keep the catch without logging?

Reply:
Ah yea, I think it should be:

```typescript
  createImageBitmap(canvas)
  .then(displayedImage =>  this.setState({ displayedImage })
  .catch(e => console.log(e));
``` 

rather than the try....catch.

then the function doesnt return a Promise, we just update the state async?

Reply:
Super! That simplifies things a lot. Thanks